plumbing/format/packfile/delta_selector_test.go

```diff
@@ -197,3 +197,8 @@ func (s *DeltaSelectorSuite) TestObjectsToPack(c *C) {
 	c.Assert(otp[2].IsDelta(), Equals, true)
 	c.Assert(otp[2].Depth, Equals, 2)
 }
+
+func (s *DeltaSelectorSuite) TestMaxDepth(c *C) {
+	dsl := s.ds.deltaSizeLimit(0, 0, 50, true)
```

Comment:
The logic to be able to create a delta with a depth of 50 is really complicated using public methods in `deltaSelector` (the only one is `ObjectsToPack()`). That's why I choose to test int using a private method.